File: src/main/java/team4/footwithme/stadium/service/response/CourtsResponse.java
package team4.footwithme.stadium.service.response;

import team4.footwithme.stadium.domain.Court;
import team4.footwithme.stadium.domain.Stadium;

public record CourtsResponse(
        Long courtId,
        Long stadiumId,
        String name
) {
    public static CourtsResponse from(Court court) {
        return new CourtsResponse(
                court.getCourtId(),
                court.getStadium().getStadiumId(),
                court.getName()
        );
    }
}

File: src/main/java/team4/footwithme/stadium/service/response/CourtsResponse.kt
package team4.footwithme.stadium.service.response

import team4.footwithme.stadium.domain.Court

@JvmRecord
data class CourtsResponse(
    val courtId: Long?,
    val stadiumId: Long?,
    val name: String?
) {
    companion object {
        fun from(court: Court?): CourtsResponse {
            return CourtsResponse(
                court!!.courtId,
                court.stadium!!.stadiumId,
                court.name
            )
        }
    }
}
